chain/network/src/routing/projected_hash_map.rs

```diff
@@ -0,0 +1,147 @@
+use near_primitives::borsh::maybestd::borrow::Borrow;
+use std::collections::HashSet;
+use std::hash::{Hash, Hasher};
+use std::marker::PhantomData;
+
+/// Wraps around `Edge` struct. The main feature of this struct, is that it's hashed by
+/// `(Edge::key.0, Edge::key.1)` pair instead of `(Edge::key.0, Edge::key.1, Edge::nonce)`
+/// triple.
+#[derive(std::cmp::Eq)]
+pub struct HashMapHelper<T, V>
+where
+    V: Borrow<T>,
+    T: Hash + PartialEq + Eq,
+{
+    inner: V,
+    pd: PhantomData<T>,
+}
+
```

Comment:
I'd keep this private with 

```diff
diff --git a/chain/network/src/routing/projected_hash_map.rs b/chain/network/src/routing/projected_hash_map.rs
index cb4ee6a8c..ba128ac43 100644
--- a/chain/network/src/routing/projected_hash_map.rs
+++ b/chain/network/src/routing/projected_hash_map.rs
@@ -6,8 +6,7 @@ use std::marker::PhantomData;
 /// Wraps around `Edge` struct. The main feature of this struct, is that it's hashed by
 /// `(Edge::key.0, Edge::key.1)` pair instead of `(Edge::key.0, Edge::key.1, Edge::nonce)`
 /// triple.
-#[derive(std::cmp::Eq)]
-pub struct HashMapHelper<T, V>
+struct HashMapHelper<T, V>
 where
     V: Borrow<T>,
     T: Hash + PartialEq + Eq,
@@ -36,6 +35,13 @@ where
     }
 }
 
+impl<T, V> Eq for HashMapHelper<T, V>
+where
+    V: Borrow<T>,
+    T: Hash + PartialEq + Eq,
+{
+}
+
 impl<T, V> Hash for HashMapHelper<T, V>
 where
     V: Borrow<T>,
@@ -69,7 +75,6 @@ impl<T, V> ProjectedHashMap<T, V>
 where
     V: Borrow<T>,
     T: Hash + PartialEq + Eq,
-    HashMapHelper<T, V>: Eq,
 {
     /// Note: We need to remove the value first.
     /// The insert inside HashSet, will not remove existing element if it has the same key.
```

As a rule of thumb, making something called `Helper` a public type signals that something's iffy going on.